includes/Services/SiteGenService.php

```diff
@@ -1011,10 +1011,16 @@ public static function get_nav_link_grammar_from_post_data( $id, $name, $url ) {
 	 * @throws Exception If there is an error during the upload process.
 	 */
 	public static function upload_images_to_wp_media_library( $image_urls ) {
-		require_once ABSPATH . 'wp-admin/includes/file.php';
 		require_once ABSPATH . 'wp-admin/includes/media.php';
 		require_once ABSPATH . 'wp-admin/includes/image.php';
 
+		global $wp_filesystem;
+
+		if ( empty( $wp_filesystem ) ) {
+			require_once ABSPATH . 'wp-admin/includes/file.php';
+			\WP_Filesystem();
+		}
+
```

Comment:
LGTM, but maybe we can reuse this one that's a bit more robust: https://github.com/newfold-labs/wp-module-onboarding-data/blob/e2d687a19e6f8d81b08e1939040267a8645c270f/includes/Services/ThemeGeneratorService.php#L14